### PR TITLE
Fix all RuntimeErrors during weights_only load from being erroneously reported with the weights_only message

### PR DESCRIPTION
**Pull request by mikaylagawarecki**
Caught in above PR #127627 

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #127627
* __->__ #132349

### CODE REVIEW

**Review by albanD — COMMENTED** [review]
Not sure what's the impact of this change tbh. Is Pickle handling this type of error specially?
Also  should we add a new unit test to ensure whatever this is fixing remains fixed?

**Comment by mikaylagawarecki** [comment]
@albanD main change is that we were always catching any error and reporting it with the `weights_only` message when using `weights_only` see [here](https://github.com/pytorch/pytorch/blob/main/torch/serialization.py#L1261-L1262) and yep sorry will add a test (I think it was tested in the above PR when I set to True, but then stopped being tested when I flipped the default back to None)

**Comment by albanD** [comment]
But aren't other things maybe going to raise this type of error as well? Why use UnpicklingError vs WeightsOnlyFailureError ?

**Comment by mikaylagawarecki** [comment]
Hm not sure how else would an `UnpicklingError` get raised since we re-implemented that entire `Unpickler` and don't inherit from `pickle.Unpickler`, but fair enough, perhaps there's something I missed

**Comment by albanD** [comment]
Ho that sounds fair then. Feel free to keep it as is then!

**Review by albanD — APPROVED** [review]
SGTM!

File: test/test_serialization.py [hunk 1]
```diff
@@ -377,11 +377,12 @@ def __reduce_ex__(self, proto):
 
         with tempfile.NamedTemporaryFile() as f:
             torch.save({"spoofed": TensorSerializationSpoofer(x)}, f)
-            f.seek(0)
-            with self.assertRaisesRegex(
-                    RuntimeError,
-                    "size is inconsistent with indices"):
-                y = torch.load(f)
+            for weights_only in (False, True):
```
Comment (mikaylagawarecki) [inline test/test_serialization.py:380]: Can't use @parametrize here as this test is in `SerializationMixin` (doesn't inherit from TestCase)

**Pull request closed without merging**